### PR TITLE
Add mode where each featurizer is applied individually

### PR DESCRIPTION
**Pull request by ml-evs**
This PR adds an option to the base featurizer to apply each sub-featurizer individually. This can be useful for timing featurisations and will open up the possibility of more flexible parallelisation (over features rather than structures).

### CODE REVIEW

**Comment by ml-evs** [comment]
@ppdebreuck this was mostly to deal with the issues raised in #126, let me know what you think! Can eventually build it out to enable distributed featurization where each process/node takes a different featurizer.

File: modnet/featurizers/featurizers.py [hunk 1]
```diff
@@ -96,6 +100,7 @@ def _fit_apply_featurizers(
         featurizers: Iterable[BaseFeaturizer],
         column: str,
         fit_to_df: bool = True,
+        mode: str = "single",
```
Comment (ppdebreuck) [inline modnet/featurizers/featurizers.py:103]: As `multi` is the default mode of the class, could we also make it default here ?

Comment (ml-evs) [reply modnet/featurizers/featurizers.py:103]: Yes, my bad

```suggestion
        mode: str = "multi",
```

**Comment by ppdebreuck** [comment]
`featurize_site` should also benefit from the modes. I see we already featurized sequentially (i.e. single mode) up to now. From what I remember, I did that for renaming - with is already handled separately in the meanwhile.

**Review by ppdebreuck — APPROVED** [review]
Thanks for the nice addition. I am strongly in favour of having a concurrent non-blocking compound per thread execution in the future.

**Pull request merged**